components/gitpod-messagebus/src/messagebus.ts

```diff
@@ -368,34 +369,35 @@ export abstract class AbstractMessageBusIntegration {
 
         await new Promise(async (resolve, reject) => {
             channel.sendToQueue(queue, content, msgOptions,
-            (err, ok) => {
-                if (!!err) {
-                    reject(err);
-                } else {
-                    resolve(ok);
-                }
-            });
+                (err, ok) => {
+                    if (!!err) {
+                        reject(err);
+                    } else {
+                        resolve(ok);
+                    }
+                });
         });
     }
 
-    protected async listen<T>(listener: MessagebusListener) {
+    protected async listen<T>(listener: MessagebusListener, token: CancellationToken): Promise<void> {
         // if we have a connection, establish a channel
         if (this.channel !== undefined) {
             await listener.establish(this.channel);
```

Comment:
probably we should pass the token down and bail out earlier if the client connection was closed